core/src/androidMain/kotlin/Peripheral.kt

```diff
@@ -113,8 +118,21 @@ public class AndroidPeripheral internal constructor(
     private val onServicesDiscovered: ServicesDiscoveredAction,
 ) : Peripheral {
 
+    private val receiver = object : BroadcastReceiver() {
+        override fun onReceive(context: Context, intent: Intent) {
+            val state = intent.getIntExtra(BluetoothAdapter.EXTRA_STATE, BluetoothAdapter.ERROR)
+            if (state == BluetoothAdapter.STATE_OFF) {
+                closeConnection()
+                _state.value = State.Disconnected()
+            }
+        }
+    }.also { applicationContext.registerReceiver(it, IntentFilter(BluetoothAdapter.ACTION_STATE_CHANGED)) }
```

Comment:
nit: We could make this slightly more readable in this class.

```suggestion
    private val receiver = onBluetoothOff {
        closeConnection()
        _state.value = State.Disconnected()
    }
```

_In another file:_

```kotlin
internal inline fun onBluetoothOff(crossinline action: () -> Unit): BroadcastReceiver {
    val receiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context, intent: Intent) {
            val state = intent.getIntExtra(BluetoothAdapter.EXTRA_STATE, BluetoothAdapter.ERROR)
            if (state == BluetoothAdapter.STATE_OFF) action.invoke()
        }
    }
    applicationContext.registerReceiver(receiver, IntentFilter(BluetoothAdapter.ACTION_STATE_CHANGED))
    return receiver
}
```